maintenance/shaka-player.md

```diff
@@ -0,0 +1,185 @@
+   difficulty of fixing conflicts or reimplementation.
+ - Non-code changes (docs, tests, demo, etc) are sometimes cherry-picked to
+   branches, depending on what the change is and whether it makes sense for
+   that branch.
+ - **Features are never cherry-picked** to release branches, as this would
+   break semantic versioning.
+
+
+## Release Policy
+
+Maintainers can make a feature release at almost any time.  There is no need to
+wait to batch up any particular number of changes or any particular significant
+features.
+
+
+## Branch Maintenance Policy
+
+We will cherry-pick bug fixes to LTS branches, plus the most recent 2 release
+branches.
+
+There are two kinds of branches we will dub LTS:
+ - The last feature release (minor release number) before a breaking release
+   (major release number) will receive fixes for 1 year
+   - _For example, v3.3 (right before v4.0) will be fixed for 1 year past the
+     release of v4.0.0 (until April 30, 2023)._
+ - The release currently used by the Cast Application Framework will continue
+   to receive fixes until CAF upgrades to a newer Shaka Player release by
+   default
+   - _As of June, 2022, this is currently v3.2._
+
+In addition to LTS branches, the most recent two branches will always receive
+fixes.  If the most recent release branch is v4.1, the most recent two branches
+would be v4.1 and v4.0.  After v4.2 is released, the most recent two branches
+would become v4.2 and v4.1.  Unless v4.0 was the current CAF branch, we would
+stop fixing v4.0 when v4.2 is released.
+
+
+## When to Make Breaking Changes
+
+We should make breaking changes infrequently if we can.  The goal should be one
+breaking release per year at most.  If changes can be made in a
+backward-compatible way (with a shim or behind a non-default configuration),
+they should be.  Then they can be released in a normal feature release.
+
+Deprecated features, shims, and temporary configurations can be removed when a
+major (breaking) release is finally made.
+
+For example, in v4.0, we were forced to make a change that disabled HLS support
+for older smart TVs.  In that same release, we removed deprecated methods,
+configurations that were added to maintain compatibility with older releases,
+etc.  Many of them had been marked deprecated for 6-12 months already.
+
+Breaking changes need to be listed in
+[`docs/tutorials/upgrade.md`](https://github.com/shaka-project/shaka-player/blob/main/docs/tutorials/upgrade.md),
+with details on what application developers will need to do to cope with each
+change.
+
+
+## Internal Release Processes
+
+After making public releases, the following internal steps are taken at Google
+to make the release available in our internal systems.  Detailed documents on
+these processes are internal-only (go/shaka-player).
+
+ 1. Update the Chromecast receiver app URLs for the Shaka Player Demo
+
+    > :pencil: **TODO**: Automate this!
```

Comment:
I just saw that this is very important because when I released v4.1 I got this error without realizing it :(

Reply:
Now that I'm looking at this again, I think there are two parts I need to mention.

One is that we need to generate a new app ID for a new feature release for compatibility.  I don't think there's a good way to automate that at the moment.  This will have to be Google's responsibility.

The other is that we need to update the registered URLs for bugfix releases for existing cast app IDs.  This is the part we could potentially "automate" through appspot, by having a URL like "v3-2-x-dot-shaka-player-demo.appspot.com" that would always point to the most recent release for the given branch.  Then that can be the registered URL for the receiver app, and it won't need to be updated in the Cast developer console on bugfix releases.

Reply:
Done.